server/src/test/java/org/elasticsearch/search/profile/AbstractProfileBreakdownTests.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search.profile;
+
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class AbstractProfileBreakdownTests extends ESTestCase {
+
+    enum TestTimingTypes {
+        ONE,
+        TWO,
+        THREE
+    }
+
+    static class TestProfileBreakdown extends AbstractProfileBreakdown<TestTimingTypes> {
+        /**
+         * Sole constructor
+         */
+        TestProfileBreakdown() {
+            super(TestTimingTypes.class);
+        }
+    }
+
+    public void testManagingDifferentTimers() {
+        TestProfileBreakdown testBreakdown = new TestProfileBreakdown();
+        Timer firstTimerOne = testBreakdown.getNewTimer(TestTimingTypes.ONE);
+        runTimer(firstTimerOne);
+        assertThat(firstTimerOne.getCount(), equalTo(1L));
+        long firstTimerOneTime = firstTimerOne.getApproximateTiming();
+
+        Timer secondTimerOne = testBreakdown.getNewTimer(TestTimingTypes.ONE);
+        runTimer(secondTimerOne);
+        // test that running second timer doesn't affect first timers count etc...
+        assertThat(firstTimerOne.getCount(), equalTo(1L));
+        assertThat(firstTimerOne.getApproximateTiming(), equalTo(firstTimerOneTime));
```

Comment:
should we check also the effect on the second timer?

Reply:
I added some additional checks